src/server/game/Scripting/ScriptMgr.h

```diff
@@ -879,6 +880,17 @@ class TC_GAME_API SceneScript : public ScriptObject
         virtual void OnSceneComplete(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) { }
 };
 
+class TC_GAME_API ConversationScript : public ScriptObject
+{
+protected:
+// Manages registration, loading, and execution of scripts.
+    ConversationScript(const char* name);
+
+public:
+
+    virtual void OnConversationCreate(Conversation* /*conversation*/, Unit* /*creator*/) { }
```

Comment:
Nice PR, but you should not rely on user calling "SetValidState", instead i think you should make "OnConversationCreate" return a boolean (default true) and set Conversation state using this boolean, so the user just have to return true or false from his script